api/server/call_logs.go

```diff
@@ -21,18 +20,9 @@ func (s *Server) handleCallLogGet(c *gin.Context) {
 		return
 	}
 
-	// TODO this API needs to change to text/plain / gzip anyway, punting
-	// optimization, but we can write this direct to the wire, too... seems like
-	// we should write some kind of writev json thing for go since we keep
-	// hitting this :(
 	var b bytes.Buffer
 	b.ReadFrom(logReader)
 
-	callObj := models.CallLog{
-		CallID:  callID,
-		AppName: appName,
-		Log:     b.String(),
-	}
-
-	c.JSON(http.StatusOK, callLogResponse{"Successfully loaded log", &callObj})
+	c.Header("Content-Type", "text/plain")
+	c.JSON(http.StatusOK, b.String())
```

Comment:
From docs, it sounds like c.JSON() is going to set the content-type to json again?

[https://godoc.org/github.com/gin-gonic/gin#Context.JSON](https://godoc.org/github.com/gin-gonic/gin#Context.JSON)

maybe c.String() or c.Data() would work better?

Reply:
Yeah, but it’s funny, line above that set content type to text worked, so the response content type at client side was text/plain, but not json